server/app/scanpy_engine/scanpy_engine.py

```diff
@@ -352,13 +394,19 @@ def annotation(self, filter, axis, fields=None):
             obs = self.data.obs[obs_selector]
             if not fields:
                 fields = obs.columns.tolist()
-            result = {"names": fields, "data": DataFrame(obs[fields]).to_records(index=True).tolist()}
+            result = {
+                "names": fields,
+                "data": DataFrame(obs[fields]).to_records(index=True).tolist(),
+            }
         else:
             var = self.data.var[var_selector]
             if not fields:
                 fields = var.columns.tolist()
-            result = {"names": fields, "data": DataFrame(var[fields]).to_records(index=True).tolist()}
-        return result
+            result = {
+                "names": fields,
+                "data": DataFrame(var[fields]).to_records(index=True).tolist(),
+            }
+        return json.dumps(result, cls=Float32JSONEncoder)
```

Comment:
Have you confirmed this will correctly throw an error if NaN is present in the data?   I am not clear if we need to specify `allow-nan=False` directly in the dumps() args, or if specifying it in the encoder is sufficient.   dumps() and jsonify() work a bit differently, so we will need to confirm so there is not a feature regression.